railties/lib/rails/tasks/framework.rake

```diff
@@ -47,8 +47,12 @@ namespace :app do
         @app_generator ||= begin
           require 'rails/generators'
           require 'rails/generators/rails/app/app_generator'
-          gen = Rails::Generators::AppGenerator.new ["rails"], { with_dispatchers: true, api: !!Rails.application.config.api_only },
-            destination_root: Rails.root
+          opts = []
+          opts << "--api" if Rails.application.config.api_only
```

Comment:
We won't need to append `--api` separately after #22790 because it will be stored in `.rails`.